tests/spec/conftest.py

```diff
@@ -97,9 +98,13 @@ def parse_row(column_types, col_names, line):
     See test_conftest.py for examples.
     """
 
+    # Regex splits on any '|' character, as long as it's not adjacent
+    # to another '|' character using look-ahead and look-behind. This
+    # is to allow '||' to appear as content within a field, currently
+    # just for the all_diagnoses and all_procedures fields in apcs
     return {
         col_name: parse_value(column_types[col_name], token.strip())
-        for col_name, token in zip(col_names, line.split("|"))
+        for col_name, token in zip(col_names, re.split(r"(?<!\|)\|(?!\|)", line))
```

Comment:
Given that this logic occurs twice in this PR and I think, as you mentioned, there's something similar going on in the in-memory database now might be the time to abstract this out into a "parse Peter's ad-hoc ascii table format" helper. Maybe something like `utils/ascii_table_utils.py`.

(Sorry that this PR seems to keep growing extra bits on it!)

Reply:
I guess my question is whether I'm breaking anything by _not_ changing the in-memory file, or I will break something by changing it. I've got a catch-up with Peter in a few minutes, so I'll pick his brains.

Reply:
Ok so the ascii parsing in `in_memory_database.py` is only used in one test file, which doesn't (and likely won't ever) contain `||`s which is why nothing breaks. I propose to merge the changes without the refactor, but to raise a ticket to pick that up in deck scrubbing.

Reply:
Sounds sensible to me.